lightblue-client-integration-test/src/test/java/com/redhat/lightblue/client/integration/test/TestLightblueExternalResource.java

```diff
@@ -3,35 +3,46 @@
 import static com.redhat.lightblue.util.test.AbstractJsonNodeTest.loadJsonNode;
 import static org.junit.Assert.assertEquals;
 
+import java.io.IOException;
+
 import org.junit.Before;
 import org.junit.ClassRule;
 import org.junit.Test;
 
 import com.fasterxml.jackson.databind.JsonNode;
-import com.redhat.lightblue.client.LightblueClient;
 import com.redhat.lightblue.client.Projection;
 import com.redhat.lightblue.client.integration.test.LightblueExternalResource.LightblueTestMethods;
 import com.redhat.lightblue.client.integration.test.pojo.Country;
 import com.redhat.lightblue.client.request.data.DataInsertRequest;
 import com.redhat.lightblue.client.response.LightblueDataResponse;
+import com.redhat.lightblue.rest.integration.FakeIdentityManager;
 
 public class TestLightblueExternalResource {
 
-    private LightblueClient client;
-
     @ClassRule
     public static LightblueExternalResource lightblue = new LightblueExternalResource(new LightblueTestMethods() {
 
         @Override
         public JsonNode[] getMetadataJsonNodes() throws Exception {
-            return new JsonNode[]{loadJsonNode("./metadata/country.json")};
+            return new JsonNode[]{
+                    loadJsonNode("./metadata/country.json")
+            };
         }
 
     });
 
     @Before
-    public void before() {
-        client = lightblue.getLightblueClient();
+    public void before() throws IOException, InterruptedException {
+        lightblue.getControllerInstance().cleanupMongoCollections("country");
+
+        if (lightblue.getControllerInstance().getIdentityManager() != null) {
+            lightblue.changeIdentityManager(null);
+
+            //TODO remove sleep. There is some sort of timing issue with the server restart.
+            //DeploymentManager might be the key, but no way to get access to it today
+            // https://github.com/undertow-io/undertow/blob/master/examples/src/main/java/io/undertow/examples/servlet/ServletServer.java
+            Thread.sleep(5000);
```

Comment:
This sleep is only needed if you're changing identity manager, right? Maybe put it in changeIdentityManager?